homeassistant/components/sunweg/__init__.py

```diff
@@ -103,7 +103,7 @@ def get_api_value(
     def get_data(
         self,
         entity_description: SunWEGSensorEntityDescription,
```

Comment:
This design isn't good. The sensor description is a platform detail and should stay within the platform. At minimum the sensor description details should be separated from the other description attributes in this signature.